sdk/core/azure-core/src/http/retry_policy.cpp

```diff
@@ -147,17 +147,14 @@ std::unique_ptr<RawResponse> RetryPolicy::Send(
 
       // If we are out of retry attempts, if a response is non-retriable (or simply 200 OK, i.e
       // doesn't need to be retried), then ShouldRetryOnResponse returns false.
-      if (!ShouldRetryOnResponse(*response.get(), m_retryOptions, attempt, retryAfter))
-      {
-        // If this is the second attempt and StartTry was called, we need to stop it. Otherwise
-        // trying to perform same request would use last retry query/headers
-        return response;
-      }
 
```

Comment:
Can we also move `WasLastAttempt` check out of `ShouldRetryOnResponse`? So that we don't need to check it again in overridden `ShouldRetry`.